fiftyone/core/odm/database.py

```diff
@@ -1484,6 +1484,98 @@ def delete_runs(name, dry_run=False):
+
+    Args:
+        dataset: the dataset to query
+        field_or_index: the name of the indexed field to query or the index key for a compound index
+        pipeline: an optional pipeline to apply to the collection. For performance,
+            this should not include any fields not covered by the index.
+        values_only (False): whether to remove field names from the resulting list.
+            If True, the field names are removed and only the values will be returned as a list for each sample.
+            If False, the field names are preserved and the values will be returned as a dict for each sample.
+    Returns:
+        a list of values for the specified field or index keys for each sample sorted in the same order as the index
+    """
+    if field_or_index == "id":
+        if values_only:
+            return [
+                str(doc["_id"])
+                for doc in _iter_indexed_values(
+                    dataset, field_or_index, pipeline
+                )
+            ]
+        return [
+            {"id": str(doc["_id"])}
+            for doc in _iter_indexed_values(dataset, field_or_index, pipeline)
+        ]
+    if values_only:
+        # If values_only is True, we need to extract the values from the dict
+        if "_1_" not in field_or_index:
+            # Single field index
+            return [
+                doc[field_or_index]
+                for doc in _iter_indexed_values(
+                    dataset, field_or_index, pipeline
+                )
+            ]
+        return [
+            list(doc.values())
+            for doc in _iter_indexed_values(dataset, field_or_index, pipeline)
+        ]
+    return _iter_indexed_values(dataset, field_or_index, pipeline).to_list()
+
+
+def _iter_indexed_values(dataset, field_or_index, pipeline=None):
+    collection = dataset._sample_collection
+    if (
+        (field_or_index != "_id")
+        and (
+            field_or_index
+            not in (default_indexes := dataset._get_default_indexes())
+        )
+        and (
+            field_or_index not in (indexes := dataset.get_index_information())
+        )
+    ):
+        raise ValueError(
+            "Field '%s' is not indexed. Please create an index first or call `values()` instead"
+            % field_or_index
+        )
+    if field_or_index == "id" or field_or_index == "_id":
+        proj = {"_id": 1}
+        hint = {"_id": 1}
+    elif field_or_index in default_indexes:
+        if "_1" not in field_or_index:
+            proj = {field_or_index: 1, "_id": 0}
+            hint = {field_or_index: 1}
+        else:
+            proj = {f: 1 for f in field_or_index.strip("_1").split("_1_") if f}
+            hint = {k: v for k, v in proj.items()}
+            proj["_id"] = 0
+
+    elif field_or_index in indexes:
+        idx = indexes[field_or_index]["key"]
+        proj = {field: int(direction) for field, direction in idx}
+        hint = {k: v for k, v in proj.items()}
```

Comment:
direction doesn't have anything to do with the projection, it's just coincidental that `bool(-1) is True`.
This would probably be more semantically accurate
```suggestion
        hint = {field: int(direction) for field, direction in idx}
        proj = {k: True for k in hint.keys()}
```